retry_delay.go

```diff
@@ -100,6 +102,9 @@ func newRetryDelayStrategy(
 // Note that currentTime is passed as a parameter instead of computed by this function to guarantee predictable
 // behavior in tests.
 func (r *retryDelayStrategy) NextRetryDelay(currentTime time.Time) time.Duration {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+
 	if !r.goodSince.IsZero() && r.resetInterval > 0 && (currentTime.Sub(r.goodSince) >= r.resetInterval) {
 		r.retryCount = 0
 	}
```

Comment:
`r.jitter` read/write is protected by the lock here, but unprotected in `hasJitter` - is that expected? (I didn't look into what the caller of `hasJitter` is.)

Reply:
Well, nothing ever modifies `r.jitter` after the object is created, right? It's a constructor parameter. Unfortunately Go doesn't let us declare it as really immutable, but we don't ever change it and there's no reason to ever change it. If you look at what this type is really doing, I think the semantics are fairly clear: it has some constructor parameters defining its behavior, and then it has some mutable state that includes `retryCount` and `resetInterval`.